common/components/data_editor.py

```
@@ -197,198 +294,168 @@ def data_selector(label: str, value: str, key: str, wd) -> tuple[str, bool]:
+    for k, value in [
+        (k, value)
+        for k, value in st.session_state.items()
         if (
-            key.startswith("workflow-config-")
-            and key.endswith("-data")
-            and isinstance(value, pd.DataFrame)
-            and not value.equals(data)
+            k.startswith("workflow-config-")
+            and k.endswith("-data")
+            and isinstance(value, pl.DataFrame)
+            and not f"{key}-data" == k
```

Ruff (0.8.2)

Use `f"{key}-data" != k` instead of `not f"{key}-data" == k`

Replace with `!=` operator

(SIM201)